tensorflow_addons/image/interpolate_spline_test.py

```diff
@@ -237,92 +213,81 @@ def test_nd_linear_interpolation(self):
 
         for order in (1, 2, 3):
             for reg_weight in (0, 0.01):
-                interpolator = interpolate_spline(train_points, train_values,
-                                                  query_points, order,
-                                                  reg_weight)
+                interp = self.evaluate(
+                    interpolate_spline(train_points, train_values,
+                                       query_points, order, reg_weight))
 
                 target_interpolation = tp.HARDCODED_QUERY_VALUES[(order,
                                                                   reg_weight)]
                 target_interpolation = np.array(target_interpolation)
-                with self.cached_session() as sess:
-                    interp_val = sess.run(interpolator)
-                    self.assertAllClose(interp_val[0, :, 0],
-                                        target_interpolation)
 
-    @test_utils.run_deprecated_v1
+                self.assertAllClose(interp[0, :, 0], target_interpolation)
+
     def test_nd_linear_interpolation_unspecified_shape(self):
         """Ensure that interpolation supports dynamic batch_size and
         num_points."""
         tp = _QuadraticPlusSinProblemND()
         (query_points, _, train_points,
          train_values) = tp.get_problem(dtype='float64')
 
-        # Construct placeholders such that the batch size, number of train points,
-        # and number of query points are not known at graph construction time.
         feature_dim = query_points.shape[-1]
         value_dim = train_values.shape[-1]
-        train_points_ph = tf1.placeholder(
-            dtype=train_points.dtype, shape=[None, None, feature_dim])
-        train_values_ph = tf1.placeholder(
-            dtype=train_values.dtype, shape=[None, None, value_dim])
-        query_points_ph = tf1.placeholder(
-            dtype=query_points.dtype, shape=[None, None, feature_dim])
 
         order = 1
         reg_weight = 0.01
 
-        interpolator = interpolate_spline(train_points_ph, train_values_ph,
-                                          query_points_ph, order, reg_weight)
+        # Get concrete functions such that the batch size, number of train points,
+        # and number of query points are not known at graph construction time.
+        fn = tf.function(interpolate_spline).get_concrete_function(
+            tf.TensorSpec(
+                shape=[None, None, feature_dim], dtype=train_points.dtype),
+            tf.TensorSpec(
+                shape=[None, None, value_dim], dtype=train_values.dtype),
+            tf.TensorSpec(
+                shape=[None, None, feature_dim], dtype=query_points.dtype),
+            order, reg_weight)
```

Comment:
As interpolate_spline and sparse_image_warp are internal requests from Google Team, I am not sure if we should directly decorate them with `tf.function`. So I only decorate it when tests need it.

Reply:
I think this is fine. @yhliang2018 may be able to comment if it's okay to decorate at all times.